### PR TITLE
Update KEP-349 to reflect actual state of the `WaitForPodsReady` API

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Updates KEP-349 to reflect actual state of the `WaitForPodsReady` API

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: keps/349-all-or-nothing/README.md [hunk 1]
```diff
@@ -215,6 +218,15 @@ type WaitForPodsReady struct {
...
 	TimeoutSeconds *int64 `json:"timeoutSeconds,omitempty"`
+
+	// BlockAdmission when true, cluster queue will block admissions for all
+	// subsequent jobs until the jobs reach the PodsReady=true condition.
+	// This setting is only honored when `Enable` is set to true.
+	BlockAdmission *bool `json:"blockAdmission,omitempty"`
+
+	// RequeuingStrategy defines the strategy for requeuing a Workload.
+	// +optional
+	RequeuingStrategy *RequeuingStrategy `json:"requeuingStrategy,omitempty"`
```
Comment (mimowo) [inline keps/349-all-or-nothing/README.md:229]: Could you also include the `RequeuingStrategy` type?

Comment (PBundyra) [reply keps/349-all-or-nothing/README.md:227]: Absolutely, done!

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
Thanks for the update!

**Pull request merged**